### DISCUSSION TITLE
Suggestions for Installs

### DISCUSSION THREAD

**Discussion by methanoid** [SpiralLinux/SpiralLinux-project#118] [General]
I like Spiral a LOT but some suggestions:

Add a Minimal install option (for those who don't want LibreOffice etc)

Add a script to desktop that changes you automatically from Stable to Testing or Unstable and does the upgrading of packages. Appreciate you can do it manually easily but would be a real boon to those who DO want newer packages to have that set up in one  or two clicks ;-)

**Comment by geckolinux** [comment]
Hi there, thanks a lot for the suggestions, and I'm glad that you like SpiralLinux.

> Add a Minimal install option (for those who don't want LibreOffice etc)

This would be a long-term project that would require somebody from the community to come up with a Calamares configuration to do that. I won't say it's impossible, but in general Calamares just decompresses the same image used to boot the live system and copies it to the hard drive. (Some distros do an online install where Calamares downloads a request set of packages from the online repos, but that's a lot slower and more prone to failures and I prefer to avoid that.) So a minimal install option would mean that it would actually have to run a bunch of `apt remove` commands toward the end of the installation. The package selection in SpiralLinux is already quite minimal, in fact you can see which packages have been added specifically for the desktop environment and its related GUI apps by category by downloading the "recipe" tarballs from this Github project and perusing the `./config/package-lists/spirallinux-Plasma-packages.list.chroot` file.

> Add a script to desktop that changes you automatically from Stable to Testing or Unstable and does the upgrading of packages. Appreciate you can do it manually easily but would be a real boon to those who DO want newer packages to have that set up in one or two clicks ;-)

I'm not sure if it would be a good idea to make that too easy, as Testing and Unstable are really not recommended unless the user really knows what they're doing. As it is, by following the steps in the wiki with Synaptic it's really just as easy as un-checking the repos related to Stable and then clicking the checkbox for Testing or Unstable.

Thanks again for the suggestions!

**Reply by methanoid** [reply]
I appreciate you taking time to reply and explain your (sensible) reasons

Just wanted to say I am early in my Linux journey and experimenting with base distros and DEs. I have found EVEN more to admire in Spiral - you seem to have ticked JUST the right boxes for me. Little quality of life improvements.

I've also gained confidence in scripting and could now easily uninstall whatever I might want and could script the change to Testing or Unstable but I think I am happy with Stable now anyway.

I was tempted to try your OpenSuse spin GeckoLinux but I think for now I shall be testing Arch based, Debian based, maybe Fedora based and Ubuntu (I know its Debian derived). I don't suppose you spin any of those? :-) 

But thank you again for an excellent piece of very thoughtful work.

**Reply by geckolinux** [reply]
Thanks again for your kind words!

> I shall be testing Arch based, Debian based, maybe Fedora based and Ubuntu (I know its Debian derived). I don't suppose you spin any of those?

Nope, not yet. ;-) Best wishes for your Linux experiments!

**Comment by methanoid** [comment]
Hi, hope this isnt too cheeky a question but can I build my own custom Spiral KDE ISO by just editting the files in the Github or do I need to use the Builder Edition?  I've been playing with ArcoLinux builder and enjoy assembling my own custom ISO for quick installation how I like it but I do like Spiral a lot too, so may be forced to have an Arch/Debian dual boot (cos I can!). Any clues welcome

**Reply by geckolinux** [reply]
Hi there, yep, definitely.

My little script at the top level of each archive lets you build an ISO automatically with the date provided in the variable in the script by simply running `spiral -i` .

The package list is defined in `config/package-lists/` , and overlay files go in `config/includes.chroot/` (which is the same as the root `/` of the installed system). Also default user settings go under `config/includes.chroot/etc/skel/` .

**Reply by geckolinux** [reply]
Also it should be noted that you'll need a Debian 11 system to build it, with the `live-build` package installed. I'd recommend using a cheap VPS for the job.

**Comment by methanoid** [comment]
Thanks for the replies. I guess I can build Spiral custom on Spiral... Spiralception? 👍

**Comment by artytux** [comment]
Hello
What beast you speak of Sam mmm, sparks of curiosity.

> 

    I shall be testing Arch based, Debian based, maybe Fedora based . I don't suppose you spin any of those


> Nope, not yet. ;-)

 
Regards
artytux

**Reply by methanoid** [reply]
If you like good spins of distros like this, check out Arcolinux for Arch. It's very impressive and customisable.
